test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpSource/HttpSourceAuthenticationHandlerTests.cs

```diff
@@ -534,6 +534,22 @@ public async Task SendAsync_RetryWithClonedPostRequest(HttpContent httpContent)
             Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
         }
 
+        [Fact]
+        public void Dispose_CallingDisposeWithoutArguments_DisposesInstance()
+        {
+            // Arrange
+            var packageSource = new PackageSource("http://package.source.test");
+            var clientHandler = new HttpClientHandler();
+            var credentialService = Mock.Of<ICredentialService>();
+            var handler = new HttpSourceAuthenticationHandler(packageSource, clientHandler, credentialService);
+
+            // Act
+            handler.Dispose();
+
+            // Assert
+            Assert.True(handler._isDisposed);
```

Comment:
Isn't this always true, even before the PR?

I'm not clear/convinced that this test is useful as written. Maybe if we introduced another property to verify whether the HTTP lock is disposed or not, then we can ensure that `handler.Dispose()` also disposed that? Even then, not sure if that needs a test... thoughts?

Reply:
I just want to test that disposing the object effectively disposes the inner Http lock. The Dispose pattern uses this variable and I just tested on top of that variable.

Reply:
Removed test.